src/parseConfigFile.ts

```diff
@@ -0,0 +1,51 @@
+import * as fs from "fs";
+import { Cell, CellValue, GameState } from "./gamestate";
+
+export function loadGameStateFromFile(filePath: string): GameState {
+  let data = parseConfigFile(filePath);
+  if (!Array.isArray(data)) {
+    throw Error(`Malformed config file ${filePath}`);
+  }
+  let rowsCount = data.length;
+  let gameState = { board: [] };
+  for (let row of data) {
+    gameState.board.push(parseRow(row, rowsCount, filePath));
```

Comment:
You break the Demeter law. You should build your board and finally build your gameState

Reply:
fixed